### PR TITLE
Require Model.generation_time, and related cleanups.

### PR DESCRIPTION
**Pull request by grahamgower**
Species specific subclasses of stdpopsim.Model have been removed. Addresses #243.

### CODE REVIEW

File: stdpopsim/catalog/arabidopsis_thaliana.py [hunk 1]
```diff
@@ -80,21 +80,15 @@
...
 # FIXME this documentation needs to be filled out.
-class _Durvasula2017MSMC(ArabidopsisThalianaModel):
+class _Durvasula2017MSMC(stdpopsim.Model):
     id = "SMA_1pop"
     name = "South Middle Atlas population size history"
     description = """
         Model estimated from two homozygous individuals from the South Middle Atlas
         using MSMC (TODO: more detail).
     """
+    species = _species
```
Comment (jeromekelleher) [inline stdpopsim/catalog/arabidopsis_thaliana.py]: Are we actually using the species attribute in client code? If not, I think we should drop it. We can use the ``_species`` variable to get access to the generation times, etc.

File: stdpopsim/cli.py [hunk 2]
```diff
@@ -381,7 +381,9 @@ def add_simulate_species_parser(parser, species):
     def run_simulation(args):
         if args.model is None:
             model = stdpopsim.PiecewiseConstantSize(species.population_size)
-            model.citations = species.population_size_citations
+            model.generation_time = species.generation_time
+            model.citations = species.population_size_citations + \
```
Comment (jeromekelleher) [inline stdpopsim/cli.py]: Safer to ``extend`` this list here. Perhaps we might have a citation in there all ready for the generic model.

**Comment by jeromekelleher** [comment]
Ah, excellent, thanks @grahamgower, looks like species is extraneous. OK, one more rebase and lets merge (I don't want to do it myself and force push to your master, seems like bad etiquette; probably not a great idea to use master as your feature branch though)

**Comment by grahamgower** [comment]
Ok, rebased again.

**Pull request merged**